src/commands/lint.ts

```diff
@@ -17,117 +17,182 @@
+  description = 'Identifies potential errors in your code.';
 
-  args = [
+  args: ArgDescriptor[] = [
     {
       name: 'input',
       type: String,
       alias: 'i',
       defaultOption: true,
       multiple: true,
-      description: 'Files and/or folders to lint. Exclusive. Defaults to cwd.'
-    },
-    {
-      name: 'policy',
-      type: String,
-      alias: 'p',
-      description: 'Your jsconf.json policy file.',
-      defaultValue: false
+      description: 'Files to lint. If given, these files will be the only ' +
+          'ones linted, otherwise all files in the project will be linted.'
     },
     {
-      name: 'config-file',
+      name: 'rules',
       type: String,
-      defaultValue: 'bower.json',
-      description: (
-          'If inputs are specified, look for `config-field` in this JSON file.')
-    },
-    {
-      name: 'config-field',
-      type: String,
-      defaultValue: 'main',
-      description:
-          ('If config-file is used for inputs, this field determines which ' +
-           'file(s) are linted.')
-    },
-    {
-      name: 'follow-dependencies',
-      type: Boolean,
-      description:
-          ('Follow through and lint dependencies. This is default behavior ' +
-           'when linting your entire application via the entrypoint, shell, ' +
-           'and fragment arguments.')
-    },
-    {
-      name: 'no-follow-dependencies',
-      type: Boolean,
-      description:
-          ('Only lint the files provided, ignoring dependencies. This is ' +
-           'default behavior when linting a specific list of files provided ' +
-           'via the input argument.')
+      alias: 'r',
+      multiple: true,
+      description: 'The lint rules/rule collections to apply. ' +
+          'See `polymer help lint` for a list of rules.',
     }
   ];
 
-  run(options: CommandOptions, config: ProjectConfig): Promise<any> {
-    // Defer dependency loading until this specific command is run
-    const polylint = require('polylint/lib/cli');
-
-    let lintFiles: string[] = options['input'];
-    if (!lintFiles) {
-      lintFiles = [];
-      if (config.entrypoint)
-        lintFiles.push(config.entrypoint);
-      if (config.shell)
-        lintFiles.push(config.shell);
-      if (config.fragments)
-        lintFiles = lintFiles.concat(config.fragments);
-      lintFiles = lintFiles.map((p) => p.substring(config.root.length));
-    }
+  /**
+   * TODO(rictic): things to make configurable:
+   *   - lint warning verbosity
+   *   - whether to use color (also: can we autodetect if color is supported?)
+   *   - add option for input files to polymer.json
+   *   - modules to load that can register new rules
+   *   - --watch
+   *   - --fix
+   */
+
+  async run(options: Options, config: ProjectConfig): Promise<any> {
+    // Defer dependency loading until this specific command is run.
+    const lintLib: typeof lintLibTypeOnly = require('polymer-linter');
```

Comment:
This is getting a little too long to manage, consider moving logic into a new `/lint` subdirectory to simplify the command itself into a single require/import call. (see `build/` and `init/` for examples where we've done this).